### PR TITLE
Introduction of a projection parameter for convert.py functions and ISG grid/geo conversion capability

### PR DESCRIPTION
**Pull request by jonathonpsmith**
Changes to various functions in convert.py to allow conversions by passing in an instance of a Projection object as a parameter.
Various changes to allow easy conversions between AGD66 and the ISG projection for NSW users.

- Remove the hardcoded 'proj' variable from convert.py and introduce a 'prj' parameter in the following functions:
  - psfandgridconv()
  - geo2grid()
  - grid2geo()
- Define new Projection instance 'isg' in constants.py
- Add new functionality to calculate the zone and central meridian for ISG
- Add new exceptions for invalid ISG zones (ISG is only valid in NSW)
- Add a warning for non-recommended ellipsoid/projection combinations for ISG
- Add new tests to test_convert.py to check the above new functionality.

### CODE REVIEW

**Comment by harry093** [comment]
Thanks @jonathonpsmith and @nicgowans, this looks really useful. I have just run the checks and all seems to be in order. If I don't get to this this afternoon, I will on Monday. I will push a new version to PyPI, too

**Comment by jonathonpsmith** [comment]
Thanks Craig, sounds great. Glad I could be useful!

**Comment by BatchelorJ** [comment]
Hi @jonathonpsmith and @nicgowans, this looks really great! Happy that you've taken the projection concept to the next level. A couple of comments:
- @harry093, as this change touches on some of the functions used in the GA geodesy apps, have these been tested and reviewed?
- @jonathonpsmith, are you able to add a comment and link in `geodepy.constants` where isg is defined to help users find the references @nicgowans highlighted?

**Comment by jonathonpsmith** [comment]
Hey @BatchelorJ, yep too easy.
I've added a description comment for ISG and both links that @nicgowans supplied, with a description of what each are.

I don't believe the projection parameter addition would affect any other functions, as it's defined with 'utm' as the default, which preserves the existing functionality.

File: geodepy/tests/resources/Test_Conversion_ISG_Geo.csv [hunk 1]
```diff
@@ -0,0 +1,212 @@
+ALBU,-36.079078850050,146.914301020650,
```
Comment (harry093) [inline geodepy/tests/resources/Test_Conversion_ISG_Geo.csv:1]: Do we need this many test points? Perhaps, a few from each zone would be enough?

File: geodepy/tests/resources/Test_Conversion_ISG_Grid.csv [hunk 2]
```diff
@@ -0,0 +1,212 @@
+ALBU,552,292281.220,1005905.150
```
Comment (harry093) [inline geodepy/tests/resources/Test_Conversion_ISG_Grid.csv:1]: As above, do we need this many test points?

**Review by harry093 — COMMENTED** [review]
@jonathonpsmith these changes look all in order. My only suggestion is to reduce the number of tests. If there is a good reason for having so many, I am happy to leave them in.

**Comment by harry093** [comment]
> as this change touches on some of the functions used in the GA geodesy apps, have these been tested and reviewed?

@BatchelorJ, I agree with @jonathonpsmith that these changes should not break anything as prj is an optional argument that has a default value (utm) defined.

To be safe, though, after these changes have been pushed through to PyPI, we can update the version of Geodesy on the dev branch and make sure everything is working correctly.

File: geodepy/tests/resources/Test_Conversion_ISG_Geo.csv (hunk 1 above)
Comment (jonathonpsmith) [reply geodepy/tests/resources/Test_Conversion_ISG_Geo.csv:1]: Happy to reduce the set, I was just mirroring the existing files as much as possible, so I used all the CORS in NSW for simplicity. I could reduce it to 5 per zone? Which would be around 50 marks - might not have 5 marks for 572 (Lord Howe Island).

File: geodepy/tests/resources/Test_Conversion_ISG_Grid.csv (hunk 2 above)
Comment (jonathonpsmith) [reply geodepy/tests/resources/Test_Conversion_ISG_Grid.csv:1]: See above reply.

File: geodepy/tests/resources/Test_Conversion_ISG_Geo.csv (hunk 1 above)
Comment (harry093) [reply geodepy/tests/resources/Test_Conversion_ISG_Geo.csv:1]: Gotcha! OK, no, leave it as it is for now. I'm going to merge the changes. Thanks for your work on this

**Comment by harry093** [comment]
@jonathonpsmith @nicgowans @BatchelorJ - GeodePy v0.0.28 is now up on PyPI

**Pull request merged**